crates/log/src/setup.rs

```diff
@@ -0,0 +1,215 @@
+use bevy::prelude::*;
+use flate2::{Compression, GzBuilder};
+use std::fs::File;
+use std::path::PathBuf;
+
+
+pub(crate) struct LogPlugin {
+    pub(crate) current_log_file: String,
+}
+
+#[derive(Resource)]
+pub(crate) struct CurrentLogFile {
+    pub(crate) path: String,
+}
+
+impl Plugin for LogPlugin {
+    fn build(&self, app: &mut App) {
+        app.insert_resource(CurrentLogFile {
+            path: self.current_log_file.clone(),
+        });
+        app.add_startup_system(compress_old_logs);
+    }
+}
+
+// get all log files in the /logs directory that are not the current log file then compress them and delete the originals
+fn compress_old_logs(current_log_file: Res<CurrentLogFile>) {
```

Comment:
How quickly to logs grow (e.g. KiB per minute of play time)? I would not implement the compression unless the logs get very large soon. And even in such a case, reducing some log levels might be a better solution.

Anyways, since this change is controversial, I would remove it from this PR. If we agree to do it in any form, you can open a separate follow up PR. That way, the rest will get merged sooner.

Reply:
Game startup consumes about 1MB and after playing for a while it reaches about 2MB. The compression is able to reduce file size from 1MB to around 50kb in only 2ms. 1 MB is not much but after playing hour long games with a lot going on in them I'd imagine it would become quite big. Gzip is a very low-cost compression algorithm.

I will remove this from the PR but I do think that compression is a good thing to have further discussion around in the future.